src/switcher.S

```diff
@@ -0,0 +1,134 @@
+#include "comps_offsets.h"
+
+.text
+.balign 4
+
+.global switcher_entry
+.global switch_compartment
+.global switch_compartment_end
+
+/**
+ * Entry point from user code to switcher function
+ *
+ * @param c0 DDC of switcher, containing compartment information
+ */
+.type switcher_entry, "function"
+switcher_entry:
+    mov      c29, c0
+    mov      x0, #0
+    cvtp     clr, lr
+    b        switch_compartment
+    ret      clr
+
+/** Code to perform actual switch
+ *
+ * @param x0 ID (in `comps` array) of compartment to switch to
```

Comment:
It also expects the switcher `DDC` in `c29`, and it assumes that its _address_ is set to some data area too. (The `DDC` address is usually ignored.)

Reply:
Yes, I've documented this in the heavily modified local copy I'm working on right now. Thanks for spotting that.